roop/processors/frame/face_enhancer.py

```diff
@@ -116,6 +90,53 @@ def restore_face(face_in_tensor) -> Any:
     return convert_tensor_to_image(face_in_tensor)
 
 
+def enhance_face_in_frame(face_list):
+    face_enhanced_list = []
+    for  _, cropped_face in enumerate(face_list):
+        face_in_tensor = normalize_face(cropped_face)
+        face_enhanced = restore_face(face_in_tensor)
+        face_enhanced_list.append(face_enhanced)
+    return face_enhanced_list
+
+
+def paste_face_back(temp_frame, face_enhanced_list, matrix_list):
+    target_img = temp_frame
+    for i, face_enhanced in enumerate(face_enhanced_list):
+        matrix = matrix_list[i]
+        face_enhanced_height = face_enhanced.shape[0]
+        face_enhanced_width = face_enhanced.shape[1]
+        inverse_matrix = cv2.invertAffineTransform(matrix)
+        face_enhanced = cv2.warpAffine(face_enhanced, inverse_matrix, (target_img.shape[1], target_img.shape[0]), borderValue=0.0)
+        image_white = np.full((face_enhanced_height,face_enhanced_width), 255, dtype=np.float32)
+        image_white = cv2.warpAffine(image_white, inverse_matrix, (target_img.shape[1], target_img.shape[0]), borderValue=0.0)
+        image_white[image_white>20] = 255
+        image_mask = image_white
+        mask_row_indices, mask_col_indices = np.where(image_mask==255)
+        mask_height = np.max(mask_row_indices) - np.min(mask_row_indices)
+        mask_width = np.max(mask_col_indices) - np.min(mask_col_indices)
+        mask_size = int(np.sqrt(mask_height*mask_width))
+        k_threshold = max(mask_size//10, 10)
+        kernel = np.ones((k_threshold,k_threshold),np.uint8)
+        image_mask = cv2.erode(image_mask,kernel,iterations = 1)
+        k_threshold = max(mask_size//20, 5)
+        kernel_size = (k_threshold, k_threshold)
+        blur_size = tuple(2*i+1 for i in kernel_size)
+        image_mask = cv2.GaussianBlur(image_mask, blur_size, 0)
+        image_mask /= 255
+        image_mask = np.reshape(image_mask, [image_mask.shape[0],image_mask.shape[1],1])
+        image_merged = image_mask * face_enhanced + (1-image_mask) * target_img.astype(np.float32)
+        image_merged = image_merged.astype(np.uint8)
+        target_img = image_merged
+    return target_img
+
+
+def process_frame(source_face: any, temp_frame: any) -> any:
+    face_list, matrix_list = align_warp_face(temp_frame)
+    face_enhanced_list = enhance_face_in_frame(face_list)
```

Comment:
should this be renamed to enhance_faces? as it proceeeds a list of faces (plural)